SimG4Core/SensitiveDetector/src/AttachSD.cc

```diff
@@ -1,32 +1,37 @@
-#include "SimG4Core/SensitiveDetector/interface/SensitiveDetectorPluginFactory.h"
 #include "SimG4Core/SensitiveDetector/interface/SensitiveDetector.h"
 #include "SimG4Core/SensitiveDetector/interface/SensitiveTkDetector.h"
 #include "SimG4Core/SensitiveDetector/interface/SensitiveCaloDetector.h"
 #include "SimG4Core/SensitiveDetector/interface/AttachSD.h"
+#include "SimG4Core/SensitiveDetector/interface/sensitiveDetectorMakers.h"
 
+#include "FWCore/Utilities/interface/Exception.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include <string>
 #include <sstream>
+#include <algorithm>
 
-AttachSD::AttachSD() {}
-
-AttachSD::~AttachSD() {}
-
-std::pair<std::vector<SensitiveTkDetector*>, std::vector<SensitiveCaloDetector*> > AttachSD::create(
+std::pair<std::vector<SensitiveTkDetector*>, std::vector<SensitiveCaloDetector*>> sim::attachSD(
+    const std::unordered_map<std::string, std::unique_ptr<SensitiveDetectorMakerBase>>& makers,
     const edm::EventSetup& es,
     const SensitiveDetectorCatalog& clg,
     edm::ParameterSet const& p,
     const SimTrackManager* man,
-    SimActivityRegistry& reg) const {
-  std::pair<std::vector<SensitiveTkDetector*>, std::vector<SensitiveCaloDetector*> > detList;
+    SimActivityRegistry& reg) {
+  std::pair<std::vector<SensitiveTkDetector*>, std::vector<SensitiveCaloDetector*>> detList;
   const std::vector<std::string_view>& rouNames = clg.readoutNames();
+
+  //auto makers = sim::sensitiveDetectorMakers(makerNames);
```

Comment:
Is this line just example, or what need to be done with it later?

Reply:
Ah, that is a left over and should be removed.